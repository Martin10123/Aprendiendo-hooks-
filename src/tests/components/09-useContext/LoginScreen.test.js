import { mount } from "enzyme";
import LoginScreen from "../../../components/09-useContext/LoginScreen";
import { UserContext } from "../../../components/09-useContext/UserContext";

describe("Pruebas en el componente LoginScreen", () => {
  const setUser = jest.fn();

  const wrapper = mount(
    <UserContext.Provider value={{ setUser }}>
      <LoginScreen />
    </UserContext.Provider>
  );

  test("Debe de mostrarse el componente correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe de ejecutar el setUser con el argumento esperado", () => {
    // wrapper.find("button").simulate("click");
    wrapper.find("button").prop("onClick")();
    const className = wrapper.find("button").prop("className");

    expect(setUser).toHaveBeenCalledWith({ id: 123, name: "ola" });
    expect(className).toBe("btn btn-primary");
  });
});
