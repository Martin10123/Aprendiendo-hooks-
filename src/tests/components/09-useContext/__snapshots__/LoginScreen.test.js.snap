// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente LoginScreen Debe de mostrarse el componente correctamente 1`] = `
<div>
  <h1>
    Login Screen
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
